### PR TITLE
chore(api): delete dead map-app skeleton from apps/api

### LINKED ISSUE #648: chore(api): delete dead map-app skeleton from apps/api (Hono migration phase 1)
Part of the apps/api → Hono migration epic #644. Phase 1: valuable standalone even if the migration stops here.

## Context

apps/api's skeleton was copied from apps/map and still carries page-app machinery that is dead weight for a pure API server — it misleads both humans and AI agents reading the codebase.

## Tasks — delete/fix in `apps/api`

1. **Delete inert middleware**: `proxy.ts` (Next 16's renamed middleware; its matcher targets `/admin/:path*` but the app has no `/admin` routes) and `middleware/with-admin.ts`, `middleware/with-editor.ts`, `middleware/types.ts`.
2. **Delete dead React/browser machinery**: `global-error.tsx`, `instrumentation-client.ts` (Sentry session replay — a browser feature), `sentry.edge.config.ts` if nothing else needs it.
3. **Trim `next.config.js`**: remove `images.remotePatterns`, the `/map` → `/` redirect, and Sentry `tunnelRoute` (ad-blocker circumvention for browsers). Keep `output: "standalone"`, `transpilePackages`, `serverExternalPackages`, and the libvips `outputFileTracingIncludes` hack — those die in Phase 3, not here.
4. **Rewrite `README.md`** — it currently describes the map application on port 3000. Describe what this service actually is: an oRPC server exposing RPC (`/v1`) + OpenAPI surfaces from `packages/api`, docs at `/docs`, deployed to Cloud Run.
5. **Dedupe the x-forwarded base-URL derivation** duplicated in `src/app/[[...rest]]/route.ts:41-49` and `src/app/docs/openapi.json/route.ts` into one `getBaseUrl(request)` util (it survives the migration verbatim).
6. **Drop now-unused deps** from `apps/api/package.json`: `next-auth`, and if nothing else uses them after step 2, `@testing-library/react` / `jsdom` / `@vitejs/plugin-react` (switch vitest `environment` from `jsdom` to `node` once `global-error.test.tsx` is gone).

## Acceptance criteria

- `pnpm turbo lint typecheck build test --filter=f3-api` passes with coverage thresholds intact.
- Staging deploy serves `/docs`, `/docs/openapi.json`, and oRPC/REST traffic identically (spot-check with each `Client` header value).
- README describes the API server, not the map.
- knip/dep-check reports no newly unused deps left behind.

---
*Filed by Claude Code - Fable 5 (API framework architectural assessment)*

### PR DESCRIPTION
**Pull request by BigGillyStyle**
## Overview

apps/api's skeleton was copied from apps/map (the map application) when it was created, and it never got cleaned up. This meant apps/api — a pure API server with no web pages of its own — was still carrying code meant for a page-rendering app: admin-page login-gate middleware for a page that doesn't exist, a React error screen with no pages to protect, browser-only error tracking config, and leftover map-app settings. This PR deletes all of that, so the codebase (for both human contributors and AI coding assistants) accurately reflects what apps/api actually is.

Part of the apps/api → Hono migration epic (#644), Phase 1.

Closes #648

## What changed

- Deleted the admin/editor login-gate middleware (`proxy.ts`, `middleware/with-admin.ts`, `middleware/with-editor.ts`, `middleware/types.ts` + its test) — it protected a `/admin` page route that has never existed in apps/api.
- Deleted the React error-boundary page (`global-error.tsx` + its test) — apps/api has no page/layout tree for it to catch errors from; every route here is a plain API handler, not a rendered page.
- Deleted browser-only and edge-runtime Sentry setup (`instrumentation-client.ts`, `sentry.edge.config.ts`), and removed the now-dead edge-runtime branch in `instrumentation.ts` that loaded it.
- Trimmed `next.config.js`: removed image-loading config, a leftover `/map` → `/` redirect, and an ad-blocker-circumvention route for Sentry that no longer applies once there's no browser client.
- Extracted the base-URL-detection logic that was duplicated in two route files into one shared helper (`src/lib/get-base-url.ts`).
- Removed testing packages (`@testing-library/react`, `@testing-library/jest-dom`, `jsdom`, `@vitejs/plugin-react`) that were only needed for the deleted React error-boundary test, and switched the test environment from a simulated browser (jsdom) to plain Node, matching what apps/api actually runs in production.
- **Kept `next-auth`** as a dependency, despite the original issue suggesting it be dropped — it's still required by the characterization test suite (the parity-gate tests for the in-progress Hono migration) to build realistic session cookies for testing real authentication behavior.
- Rewrote `README.md` from scratch — it previously described the map application (wrong port, wrong framework details, wrong feature list) instead of this API service.
- Added a test for the previously-untested `/docs` page, and fixed a stale copy of the removed `/map` redirect inside the characterization test harness (both found while verifying the change).

## Commands run

- `pnpm turbo lint typecheck build test --filter=f3-api` — all pass; test coverage is now 98.18% (up from the prior 91.04% threshold)
- `pnpm turbo run test:characterization --filter=f3-api` — 105/105 passing (confirms keeping `next-auth` was necessary and correct)
- `pnpm lint:unused` (knip, repo-wide) — zero findings
- `pnpm turbo format --filter=f3-api` — clean

## Impact to DB/env

None. No migrations, no new environment variables. `next-auth` and `@acme/auth` remain as dependencies; four dev-only testing packages were removed.

## Follow-up filed separately

While reading `next.config.js`, I found that apps/api's Sentry error-reporting configuration (project name and DSN) is copy-pasted from apps/map, meaning apps/api's production errors have likely never had their own dedicated Sentry project. Fixing that requires Sentry dashboard access I don't have, so I filed it separately: #863.

_written by Claude Sonnet 5_

## Summary by CodeRabbit

* **Documentation**
  * Replaced outdated map documentation with comprehensive F3 Nation API setup, authentication, testing, deployment, and reference guidance.
* **Bug Fixes**
  * Improved base URL handling for API redirects and OpenAPI documentation links across proxied and direct requests.
  * Removed obsolete map routing and redirects.
* **Testing**
  * Added coverage for API documentation, base URL behavior, and environment validation.
  * Streamlined server-side testing and increased coverage thresholds.

### CODE REVIEW

File: apps/api/README.md [hunk 1]
```diff
@@ -1,132 +1,92 @@
...
-## Environment Variables
-
-Environment variables are application-specific in this monorepo. The `.env` file should be placed in the application directory (`apps/map/`) rather than the monorepo root.
-
-Required environment variables (typically provided in env.zip):
+```bash
+curl -X GET "https://api.f3nation.com/v1/ping" \
+  -H "Authorization: Bearer YOUR_API_KEY" \
+  -H "Client: my-app"
+```
```
Comment (BigGillyStyle) [inline apps/api/README.md]: **🟠 Important** · code-reviewer

The auth exception list is incomplete, and the worked example demonstrates the gap. `/v1/ping` is a `publicProcedure` (`packages/api/src/router/ping.ts`) whose own description says "No authentication required," and the characterization golden `characterization/__snapshots__/dispatch-rest-ping.golden.json` records a `200` for a bare `/v1/ping` request with no `Authorization` or `Client` header. `/docs` and `/docs/openapi.json` are unauthenticated too.

The curl example two lines below the claim hits exactly this endpoint — so it "demonstrates" auth headers that aren't actually checked. A reader following this example will see a `200` and believe their key is valid, then get a real `401` on their first genuinely gated call.

**Suggested fix:** widen the exception list to `except /v1/ping, /v1/status, /docs, and /docs/openapi.json`, and swap the curl example to a genuinely gated endpoint (e.g. `/v1/regions`).

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 2]
```diff
@@ -1,132 +1,92 @@
...
 
-- Database connection strings
-- Authentication secrets
-- API keys
-- Application configuration
+API keys inherit the roles and permissions of their owner (**Editor** or **Admin**). Generate and manage keys at `map.f3nation.com/admin/api-keys` if you're an admin on a region or the F3 Nation organization. See `/docs` for the full auth contract, error responses, and rate limits.
```
Comment (BigGillyStyle) [inline apps/api/README.md]: **🟡 Suggestion** · comment-analyzer

The API-key management URL is wrong. The api-keys page lives in `apps/admin` (`apps/admin/src/app/api-keys/page.tsx`, route `/api-keys`, no `/admin` prefix), a separate deployable app served at `admin.f3nation.com` per `.github/workflows/deploy-admin.yml`. `map.f3nation.com/admin/api-keys` isn't where this page is.

```suggestion
API keys inherit the roles and permissions of their owner (**Editor** or **Admin**). Generate and manage keys at `admin.f3nation.com/api-keys` if you're an admin on a region or the F3 Nation organization. See `/docs` for the full auth contract, error responses, and rate limits.
```

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 3]
```diff
@@ -1,132 +1,92 @@
...
-This application is built with:
+## Deployment
 
-- **Next.js 14** with App Router
-- **React** with TypeScript
-- **Tailwind CSS** for styling
-- **oRPC** for type-safe API calls
-- **Drizzle ORM** for database operations
-- **Vitest** for unit testing
+Deployed to GCP Cloud Run via tag-based deploys (tag `api@X.Y.Z` on `main` triggers `.github/workflows/deploy-api.yml`), same pattern as [`apps/me`](../me/README.md#deployment). Staging and production run as separate Cloud Run services (`f3-api-app-staging`, `f3-api-app`).
```
Comment (BigGillyStyle) [inline apps/api/README.md]: **🟡 Suggestion** · code-reviewer

This conflates GCP project IDs with Cloud Run service names. In `.github/workflows/deploy-api.yml`, `f3-api-app-staging`/`f3-api-app` are the `staging_project`/`prod_project` inputs (GCP project IDs); the Cloud Run **service** is `f3-api` in both environments — same service, two different projects. `apps/me/README.md`'s own deployment table makes exactly this distinction (Cloud Run Service `f3-me` for both environments, only the GCP Project column differs), so this line's "same pattern as apps/me" claim doesn't match what apps/me's own README documents.

```suggestion
Deployed to GCP Cloud Run via tag-based deploys (tag `api@X.Y.Z` on `main` triggers `.github/workflows/deploy-api.yml`), same pattern as [`apps/me`](../me/README.md#deployment). Staging and production deploy the same Cloud Run service (`f3-api`) into separate GCP projects (`f3-api-app-staging`, `f3-api-app`).
```

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 4]
```diff
@@ -1,132 +1,92 @@
...
-Materialized views (or live joins)
-⁃ region with location summary
-⁃ maps (get all lat,lngs with some supplemental data)
-⁃ maps (get data for a particular location)
-⁃ Continue conversation on slack
-⁃ Slackbot: f3-nation-slack-bot/docs/api/endpoint_requirements.md at main · F3-Nation/f3-nation-slack-bot · GitHub
-
-API Key can be for a region or a userId
+- [Main Monorepo README](../../README.md) — overview of the entire monorepo structure
+- [API Package README](../../packages/api/README.md) — router/procedure implementation
```
Comment (BigGillyStyle) [inline apps/api/README.md]: **🟡 Suggestion** · comment-analyzer

This link is broken — `packages/api/README.md` doesn't exist (never has). Only `packages/api/AGENTS.md` and `packages/api/CHANGELOG.md` exist in that directory.

```suggestion
- [API Package AGENTS.md](../../packages/api/AGENTS.md) — router/procedure implementation
```

_written by Claude Sonnet 5_

File: apps/api/src/lib/get-base-url.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/** Derives the externally-visible base URL from proxy headers, falling back to the request's own URL. */
```
Comment (BigGillyStyle) [inline apps/api/src/lib/get-base-url.ts]: **🟡 Suggestion** · comment-analyzer

This comment describes only the fallback path and omits the function's actual highest-priority source: `envBase` (`NEXT_PUBLIC_API_URL`) wins over everything when set — the proxy-header/request-URL derivation only runs when it's unset. Since `.env.example` sets `NEXT_PUBLIC_API_URL`, the header-parsing logic the comment centers on is effectively dead in normal deployments, which the comment doesn't convey.

```suggestion
/** Base URL for the API: `NEXT_PUBLIC_API_URL` if set, else derived from proxy headers (`x-forwarded-proto`/`x-forwarded-host`) or the request's own URL. */
```

_written by Claude Sonnet 5_

File: apps/api/__tests__/docs-route.test.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+describe("docs route", () => {
+  it("returns the Scalar API reference page", async () => {
+    const { GET } = await import("../src/app/docs/route");
+
+    const response = await GET();
+
+    expect(response.status).toBe(200);
+    expect(response.headers.get("content-type")).toContain("text/html");
+    const body = await response.text();
+    expect(body).toContain("F3 Nation API Reference");
```
Comment (BigGillyStyle) [inline apps/api/__tests__/docs-route.test.ts:12]: **🟠 Important** · code-reviewer

This test covers the route without protecting its one piece of variable behavior. `src/app/docs/route.ts` has exactly one non-literal input: `baseServerURL: env.NEXT_PUBLIC_API_URL`. `pageTitle` is a hardcoded string literal in the same file, so `expect(body).toContain("F3 Nation API Reference")` is self-referential — it passes whether or not `baseServerURL` is wired at all. The value is observable (the rendered HTML embeds it in the Scalar config blob), so a real assertion is cheap to add.

If `baseServerURL` gets dropped or misrouted during the Hono migration, the Scalar "Try it" button would start firing requests at the wrong origin, and this test (plus the file's 100% line coverage) would stay green throughout.

**Suggested fix:** add `expect(body).toContain(process.env.NEXT_PUBLIC_API_URL!)` (or import `env` from `~/env` and assert against `env.NEXT_PUBLIC_API_URL`).

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 7]
```diff
@@ -1,132 +1,92 @@
...
+| Hosting   | GCP Cloud Run (via GitHub Actions)        |
 
-3. **Environment setup**:
-   - Get env.zip from F3 Nation Slack
-   - Unzip and rename to `.env`
-   - Place the `.env` file in this directory (`apps/map/.env`)
+## Authentication
 
-4. **Start development server**:
+All API endpoints (except `/v1/status`) require a Bearer API key and a `Client` header identifying the calling app:
```
[the remark this answers is not in the document]
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — widened the exception list to `/v1/ping`, `/v1/status`, `/docs`, and `/docs/openapi.json`, documented session-cookie/JWT as auth alternatives, and swapped the curl example to `/v1/org` (a genuinely `protectedProcedure`-gated endpoint — `/v1/regions` doesn't exist; the mounted prefix is `/org`).

_written by Claude Sonnet 5_

File: apps/api/README.md (hunk 1 above)
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — widened the exception list to `/v1/ping`, `/v1/status`, `/docs`, and `/docs/openapi.json`, documented session-cookie/JWT as auth alternatives, and swapped the curl example to `/v1/org` instead of `/v1/regions` — I verified `/v1/regions` doesn't actually exist; the mounted prefix is `/org` (see `packages/api/src/index.ts`).

_written by Claude Sonnet 5_

File: apps/api/README.md (hunk 2 above)
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — corrected to `admin.f3nation.com/api-keys`.

_written by Claude Sonnet 5_

File: apps/api/README.md (hunk 3 above)
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — corrected to describe one Cloud Run service (`f3-api`) deployed into two GCP projects, matching apps/me's documented pattern. Verified against `.github/workflows/deploy-api.yml`.

_written by Claude Sonnet 5_

File: apps/api/README.md (hunk 4 above)
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — repointed to `packages/api/AGENTS.md`, which is the file that actually exists.

_written by Claude Sonnet 5_

File: apps/api/src/lib/get-base-url.ts (hunk 5 above)
Comment (BigGillyStyle) [reply apps/api/src/lib/get-base-url.ts]: Fixed — the comment now leads with `NEXT_PUBLIC_API_URL` precedence before the fallback derivation.

_written by Claude Sonnet 5_

File: apps/api/__tests__/docs-route.test.ts (hunk 6 above)
Comment (BigGillyStyle) [reply apps/api/__tests__/docs-route.test.ts:12]: Fixed — added an assertion on the embedded `baseServerURL`, so the test now actually fails if it's dropped or misrouted.

_written by Claude Sonnet 5_

File: apps/api/__tests__/env.test.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  it("falls through to npm_lifecycle_event when CI and SKIP_ENV_VALIDATION are unset", async () => {
+    vi.stubEnv("CI", "");
+    vi.stubEnv("SKIP_ENV_VALIDATION", "");
+    vi.stubEnv("npm_lifecycle_event", "lint");
+    vi.resetModules();
+
+    const { env } = await import("../src/env");
+
+    expect(env).toBeDefined();
```
[the remark this answers is not in the document]
Comment (BigGillyStyle) [reply apps/api/__tests__/env.test.ts:37]: Fixed — stubbed the three required vars (`NEXT_PUBLIC_API_URL`, `NEXT_PUBLIC_CHANNEL`, `AUTH_SECRET`) to invalid values before each import, so the test now fails if the corresponding skip-validation branch isn't actually honored, instead of trivially passing against the local `.env` values.

_written by Claude Sonnet 5_

File: apps/api/src/lib/get-base-url.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+export function getBaseUrl(request: Request): string {
+  const envBase = process.env.NEXT_PUBLIC_API_URL ?? undefined;
+  const forwardedProto = request.headers.get("x-forwarded-proto") ?? undefined;
+  const forwardedHost = request.headers.get("x-forwarded-host") ?? undefined;
+  const url = new URL(request.url);
+  const host = forwardedHost ?? request.headers.get("host") ?? url.host;
+  const proto = forwardedProto ?? url.protocol.replace(":", "");
+  const derivedBase = `${proto}://${host}`;
+  return (envBase ?? derivedBase).replace(/\/$/, "");
```
[the remark this answers is not in the document]
Comment (BigGillyStyle) [reply apps/api/src/lib/get-base-url.ts]: Split reply for the two findings in this thread:

- **Empty configured URL**: fixed. A blank or whitespace-only `NEXT_PUBLIC_API_URL` is now treated as unset, falling through to the header/request-URL derivation instead of producing an empty base URL. Added `apps/api/__tests__/get-base-url.test.ts` to cover this directly.
- **Make `NEXT_PUBLIC_API_URL` mandatory / reject forwarded-header fallback**: pushing back. Issue #648 task 5 explicitly scoped this helper to preserve the existing forwarded-header fallback behavior verbatim while centralizing it during the migration — rejecting that fallback outright is a real behavior change (it would break local/branch deploys that don't set the env var) and is out of scope for this docs/cleanup PR. Open to a follow-up hardening issue if there's appetite for it.

_written by Claude Sonnet 5_

[the remark this answers is not in the document]
Comment (BigGillyStyle) [reply apps/api/src/lib/get-base-url.ts:14]: Existing indirect coverage in `rest-route.test.ts`/`openapi-route.test.ts` already exercises configured-URL precedence, forwarded headers, request-URL fallback, and trailing-slash normalization for `getBaseUrl`. Added a dedicated `apps/api/__tests__/get-base-url.test.ts` for the one real gap — the empty/whitespace-only-string edge case from the sibling comment on this line.

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 10]
```diff
@@ -1,132 +1,92 @@
...
+# Copy and populate env file
+cp .env.example .env
+# Edit .env with actual values (get from team via Slack, or use local defaults for Docker dev)
+
+# Install dependencies (from monorepo root)
+cd ../..
+pnpm install
+
+# Run the dev server
+pnpm dev --filter f3-api
```
[the remark this answers is not in the document]
Comment (BigGillyStyle) [reply apps/api/README.md:59]: This is a false positive for this repo — `pnpm dev` here is a custom root script (`scripts/dev.sh`) that explicitly parses and forwards `--filter` to `turbo dev`, not native pnpm workspace filtering. This exact form (`pnpm dev --filter <app>`) matches the documented convention in `apps/me/README.md` and `apps/map/README.md`. No change made.

_written by Claude Sonnet 5_

**Comment by BigGillyStyle** [comment]
Re: the Linked Issues check flagging that `next-auth` is still in `apps/api/package.json` — verified it: `characterization/fixtures/cookies.ts` still imports `next-auth/jwt` to encode session cookies for the characterization suite, so it's genuinely needed there, but no production runtime code in `apps/api` imports it. Moved it from `dependencies` to `devDependencies` (and updated the lockfile) to satisfy issue #648's intent without breaking the test suite — it was never in the production `.next/standalone` output either way, since Next traces actual imports rather than the dependency-section listing.

_written by Claude Sonnet 5_

File: apps/api/README.md [hunk 11]
```diff
@@ -1,132 +1,96 @@
...
+curl -X GET "https://api.f3nation.com/v1/org" \
+  -H "Authorization: Bearer YOUR_API_KEY" \
+  -H "Client: my-app"
+```
 
-- Database connection strings
-- Authentication secrets
-- API keys
-- Application configuration
+API keys inherit the roles and permissions of their owner (**Editor** or **Admin**). Generate and manage keys at `admin.f3nation.com/api-keys` if you're an admin on a region or the F3 Nation organization. See `/docs` for the full auth contract, error responses, and rate limits.
```
Comment (taterhead247) [inline apps/api/README.md]: That's not true. API keys have their own role mapping table. The association with an owner is just so we know who to blame.

**Review by taterhead247 — APPROVED** [review]
I tested api docs, map, admin, and me locally. No issues.

File: apps/api/README.md (hunk 11 above)
Comment (BigGillyStyle) [reply apps/api/README.md]: Fixed — you're right. Verified in `packages/db/drizzle/schema.ts` (`rolesXApiKeysXOrg` is keyed on `apiKeyId`, not the owner) and `packages/api/src/shared.ts`'s `getSession` (roles are looked up by `apiKeyId`, never via the owner's `userId`). Corrected the README to describe the key's own per-org role assignments instead of inheritance from the owner.

_written by Claude_
